src/Build/Logging/ParallelLogger/ConsoleOutputAligner.cs

```diff
@@ -0,0 +1,148 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#nullable enable
+using System;
+using System.Text;
+using Microsoft.Build.Shared;
+
+namespace Microsoft.Build.BackEnd.Logging
+{
+    /// <summary>
+    /// Align output to multiple lines so no logged test is lost due to limited <see cref="Console.BufferWidth"/>.
+    /// During alignment optional prefix/indent is applied.
+    /// </summary>
+    /// <remarks>
+    /// This class is not thread safe.
+    /// </remarks>
+    internal class ConsoleOutputAligner
+    {
+        internal const int ConsoleTabWidth = 8;
+
+        private readonly StringBuilder _reusedStringBuilder = new(1024);
```

Comment:
Isn't there some kind of central SB pooling you can use? I can't search the code at the moment.

Reply:
I am aware of existence of their pool, but since I am optimizing for diag and it such mode usage of this StringBuilder is quite hot, there is no point, IMHO, to rent and return sb to the pool when its constantly needed.